jetty-http/src/main/java/org/eclipse/jetty/http/pathmap/BasePathSpec.java

```diff
@@ -0,0 +1,114 @@
+//
+//  ========================================================================
+//  Copyright (c) 1995-2020 Mort Bay Consulting Pty Ltd and others.
+//  ------------------------------------------------------------------------
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.http.pathmap;
+
+/**
+ * The base PathSpec, what all other path specs are based on
+ */
+public abstract class BasePathSpec implements PathSpec
```

Comment:
I'm not sure it is worthwhile having `BasePathSpec` and `Decomposition`.

I understand they exist to try and achieve some code reuse between the 3 implementations, but ultimately you are creating two new classes just to save duplicating a few simple getters.

Perhaps a compromise is to keep `Decomposition` available as a getter on the interface, and then have default methods for the ohter getters.    Or just have an abstract base class and no interface.  Or just duplicate the fields/getters in the impls.

I'd probably like `Decomposition` better if it had a better name.... `PathSpecComponets`??? nah still horrid....

Reply:
`BasePathSpec` doesn't do much, but there's a little more in common between `RegexPathSpec` and `UriTemplatePathSpec`.

It still doesn't justify the `Decomposition`, though so I'll also duplicate that around.